src/util/sharing_map.h

```diff
@@ -445,140 +459,138 @@ SHARING_MAPT(void)::get_delta_view(
+      const innert *cp2 = static_cast<const innert *>(bp2);
+
+      const leaf_listt &ll1 = cp1->get_container();
+
+      for(const auto &l1 : ll1)
       {
         const key_type &k1=l1.get_key();
-        bool found=false;
+        const leaft *p;
 
-        for(const auto &l2 : n2->get_container())
-        {
-          const key_type &k2=l2.get_key();
+        p = cp2->find_leaf(k1);
 
-          if(l1.shares_with(l2))
-          {
-            found=true;
-            break;
-          }
-
-          if(key_equal()(k1, k2))
+        if(p != nullptr)
+        {
+          if(!l1.shares_with(*p))
           {
-            delta_view.push_back(
-              delta_view_itemt(
-                true,
-                k1,
-                l1.get_value(),
-                l2.get_value()));
-
-            found=true;
-            break;
+            delta_view.push_back({true, k1, l1.get_value(), p->get_value()});
           }
         }
-
-        if(!only_common && !found)
+        else if(!only_common)
         {
-          delta_view.push_back(
-            delta_view_itemt(
-              false,
-              l1.get_key(),
-              l1.get_value(),
-              dummy));
+          delta_view.push_back({false, l1.get_key(), l1.get_value(), dummy});
         }
       }
     }
-    else
-    {
-      UNREACHABLE;
-    }
   }
   while(!stack.empty());
 }
 
-SHARING_MAPT2(, node_type *)::get_container_node(const key_type &k)
+SHARING_MAPT2(, innert *)::get_container_node(const key_type &k)
 {
-  size_t key=hash()(k);
-  node_type *p=&map;
+  std::size_t key = hash()(k);
+  innert *ip = &map;
+
+  std::size_t bit;
 
   for(std::size_t i = 0; i < steps; i++)
   {
-    std::size_t bit = key & mask;
-    key>>=chunk;
+    bit = key & mask;
 
-    p=p->add_child(bit);
+    ip = ip->add_child(bit);
+
+    key >>= chunk;
   }
 
-  return p;
+  return ip;
 }
 
-SHARING_MAPT2(const, node_type *)::get_container_node(const key_type &k) const
+SHARING_MAPT2(const, innert *)::get_container_node(const key_type &k) const
 {
-  size_t key=hash()(k);
-  const node_type *p=&map;
+  if(empty())
+    return nullptr;
+
+  std::size_t key = hash()(k);
+  const innert *ip = &map;
+
+  std::size_t bit;
 
   for(std::size_t i = 0; i < steps; i++)
   {
-    std::size_t bit = key & mask;
-    key>>=chunk;
+    bit = key & mask;
 
-    p=p->find_child(bit);
-    if(p==nullptr)
+    ip = ip->find_child(bit);
+    if(ip == nullptr)
       return nullptr;
-  }
 
-  assert(p->is_container());
+    key >>= chunk;
+  }
 
-  return p;
+  return ip;
 }
 
-SHARING_MAPT2(const, node_type *)::get_leaf_node(const key_type &k) const
+SHARING_MAPT2(const, leaft *)::get_leaf_node(const key_type &k) const
 {
-  const node_type *p=get_container_node(k);
-  if(p==nullptr)
+  const innert *cp = get_container_node(k);
+  if(cp == nullptr)
     return nullptr;
 
-  p=p->find_leaf(k);
+  const leaft *lp;
+  lp = cp->find_leaf(k);
```

Comment:
not sure that separating declaration and initialisation improves readability here

Reply:
Ok, I'll pull this method up into the class.